primitives/runtime/src/generic/mod.rs

```diff
@@ -44,7 +44,7 @@ fn encode_with_vec_prefix<T: Encode, F: Fn(&mut Vec<u8>)>(encoder: F) -> Vec<u8>
 	let size = ::sp_std::mem::size_of::<T>();
 	let reserve = match size {
 		0..=0b00111111 => 1,
-		0..=0b00111111_11111111 => 2,
+		0b01000000..=0b00111111_11111111 => 2,
```

Comment:
I don't believe there's cast iron guarentees on arm ordering for match statements so altering this to make sure it does not overlap.